### DISCUSSION TITLE
bug: Passing `0n` to `gasPrice` caused failed transaction in MetaMask

### DISCUSSION THREAD

**Discussion by oxSaturn** [wevm/viem#2068] [Question]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

2.9.8

### Current Behavior

We had an interesting bug in our code where we pass `0n` to `gasPrice` in `prepareTransactionRequest`:

```
prepareTransactionRequest({ gasPrice: BigInt("") });
```

Then MetaMask would fail to populate a gas price while wallets like rabby has no such issue:

![CleanShot-8AfP5MiN@2x](https://github.com/wevm/viem/assets/126733611/f7c68a0b-759b-4ac2-b67d-2de4c730d516)

I just checked the source code here https://github.com/wevm/viem/blob/03095724427a0662694801392d38d6a7b5dd43c7/src/actions/public/estimateFeesPerGas.ts#L171

```
const gasPrice =
    request?.gasPrice ??
    multiply(await getAction(client, getGasPrice, 'getGasPrice')({}))
```

Seems it would take the `0n` as gasPrice.

### Expected Behavior

I'm not sure if viem should handle the `0n` case, probably not, then a warning might be helpful?

**Comment by jxom** [comment]
Hmm yeah, I would rather not be opinionated about this, because maybe some chain might (arguably) use a gasPrice of zero. Best to resolve the bug upstream.
